pkg/querier/batch/merge.go

```diff
@@ -52,6 +61,29 @@ func newMergeIterator(cs []GenericChunk) *mergeIterator {
 	return c
 }
 
+func (c *mergeIterator) Reset(size int) *mergeIterator {
+	c.its = c.its[:0]
+	c.h = c.h[:0]
+	c.batches = c.batches[:0]
+
+	if len(c.its) <= cap(c.batchesBuf) {
```

Comment:
Is this always true? As you just reset `c.its = c.its[:0]`.

Reply:
I dont think so, i think some series can have more chunks than others on the same query, and so, this would not be true, right?

Reply:
ok.. i know see what u said.. the check should check the 'size' received as parameter... 

It was not a problem as we check the something before calling reset... but for the sake of make the method safe, lets check here again.